WalletWasabi/Blockchain/Analysis/Clustering/SmartLabel.cs

```diff
@@ -1,153 +1,172 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace WalletWasabi.Blockchain.Analysis.Clustering;
 
-public class SmartLabel : IEquatable<SmartLabel>, IEquatable<string>, IEnumerable<string>, IComparable<SmartLabel>
+public readonly struct SmartLabel : IEquatable<SmartLabel>, IComparable<SmartLabel>, IReadOnlyCollection<string>
 {
+	private readonly string[]? _labels;
+	
 	public SmartLabel(params string[] labels) : this(labels as IEnumerable<string>)
 	{
 	}
 
-	public SmartLabel(IEnumerable<string> labels)
+	public SmartLabel(IEnumerable<string>? labels)
```

Comment:
The original code assumes that even if the argument isn't nullable, it's possible to get it:

https://github.com/zkSNACKs/WalletWasabi/blob/eb062deac486200080855bb70dcfd12cc90c23dc/WalletWasabi/Blockchain/Analysis/Clustering/SmartLabel.cs#L13-L15

Therefore, it's correct to allow a null reference to be passed. Otherwise, there should be an exception since `null` isn't allowed in case of removal of the nullability marker.